swarm/network/stream/peer.go

```diff
@@ -58,6 +58,7 @@ var ErrMaxPeerServers = errors.New("max peer servers")
 // Peer is the Peer extension for the streaming protocol
 type Peer struct {
 	*protocols.Peer
+	bzzPeer  *network.BzzPeer
```

Comment:
The `bzzPeer` has access to the `protocols.Peer`, so maybe we don't need the embedding on line 60 ?

Reply:
I applied a change to address this which resulted in quite a lot of cascading changes...I hope this does not introduce some new level of "ugliness"?

Reply:
@holisticode all the `peer` structs are confusing to me. I suggested the change in case it didn't result in many cascading changes. If this is not the case, feel free to just address this particular problem, and once this is correct, we can review and discuss how to refactor :)

Reply:
@holisticode I am sorry if you have understood my comment as a request to do that change - it was not intended as such.

I think the new cascading change is much worse to what we have right now, and will also result in many conflicts with other branches we have in progress right now.

Furthermore this cascading change is not addressing the underlying problem - which is the mess of all the different `peer` structs and their inheritance.

---

I suggest we just fix the proximity calculations if possible, and not do such cascading change throughout the codebase, unless we actually address the underlying problem of the design.